Excel.TemplateEngine/FileGenerating/Primitives/IExcelDocument.cs

```diff
@@ -34,5 +34,9 @@ public interface IExcelDocument : IDisposable
         void AddDescription([NotNull] string text);
 
         void CopyVbaInfoFrom([NotNull] IExcelDocument excelDocument);
+
+        bool TryGetCustomProperty(string key, [CanBeNull] out string value);
```

Comment:
тут будет правильнее заюзать атрибут `ContractAnnotation`:
```suggestion
        [ContractAnnotation("=> true, value:notnull; => false, value:null")]
        bool TryGetCustomProperty([NotNull] string key, out string value);
```

ну и у `SetCustomProperty` надо разметить тоже